tests/app/shared/issue/assignee/assignee.component.spec.ts

```diff
@@ -0,0 +1,138 @@
+    .overrideProvider(IssueService, { useValue: issueService })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AssigneeComponent);
+    component = fixture.componentInstance;
+
+    component.team = dummyTeam;
+    component.issue = dummyIssue;
+    fixture.detectChanges();
+    debugElement = fixture.debugElement;
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should have a placeholder value of - given no assignees', () => {
+    const matPlaceholderValue: HTMLElement = debugElement.query(By.css('.mat-select-placeholder')).nativeElement;
+    expect(matPlaceholderValue).toBeDefined();
+    expect(matPlaceholderValue.innerText).toEqual('-'); // Placeholder Value
+  });
+
+  it('should be able to open the assignee selector', () => {
+    const matSelect: HTMLElement = debugElement.query(By.css('.mat-select-trigger')).nativeElement;
+
+    // Open the assignee selector
+    matSelect.click();
+    fixture.detectChanges();
+    const matOption: HTMLElement = debugElement.query(By.css('.mat-option')).nativeElement;
+    const inputElement: HTMLElement = debugElement.query(By.css('.mat-select-panel')).nativeElement;
+
+    expect(inputElement.children.length).toBe(dummyTeam.teamMembers.length);
+    expect(matOption.attributes.getNamedItem('aria-selected').value).toEqual('false');
+  });
+
+  it('should emit the issueUpdated event upon closing the MatSelect', () => {
+    spyOn(component.issueUpdated, 'emit');
+    const matSelect: HTMLElement = debugElement.query(By.css('.mat-select-trigger')).nativeElement;
+
+    // Open the assignee selector
+    matSelect.click();
+    fixture.detectChanges();
+
+    // Close the matSelect element and make relevant fake calls.
+    const matSelectElement = debugElement.query(By.css('.mat-select')).nativeElement;
+    issueService.updateIssue.and.callFake(() => of(dummyIssue));
+    matSelectElement.dispatchEvent(new Event('closed'));
+    fixture.detectChanges();
+
+    expect(component.issueUpdated.emit).toHaveBeenCalledTimes(1);
+  });
+
+  it('should show the new assignee value upon adding a new assignee', () => {
+    const matSelect: HTMLElement = debugElement.query(By.css('.mat-select-trigger')).nativeElement;
+
+    // Open the Assignee Selector
+    matSelect.click();
+    fixture.detectChanges();
+    const matOption: HTMLElement = debugElement.query(By.css('.mat-option')).nativeElement;
+
+    // Assign a new assignee
+    matOption.click();
+    fixture.detectChanges();
+    expect(matOption.attributes.getNamedItem('aria-selected').value).toEqual('true');
+
+    // Close the matSelect element and make relevant fake calls.
+    const matSelectElement = debugElement.query(By.css('.mat-select')).nativeElement;
+    issueService.updateIssue.and.callFake(() => of(dummyIssue));
+    matSelectElement.dispatchEvent(new Event('closed'));
+    component.issue.assignees = [ testStudent.loginId ];
```

Comment:
In this test, let's focus on verifying that the `mat-list` correctly reflects the `assignees` of the issue set as the `AssigneeComponent`'s `Input` (i.e. `component.issue`).

Hence, there's no need to open `mat-select` and select the `mat-option`. We can skip straight to setting `component.issue.assignees`

Reply:
Ah I see, I thought we would want to simulate the entire flow of adding the assignee via the interface. In this case, i will just modify the attributes directly!

Reply:
Sure, but the the "entire flow of adding the assignee" is composed of two independent steps in the AssigneeComponent's contract:

- User selects an assignee option --> AssigneeComponent updates the issue's assignees and emits the updated issue to its parent component
- Parent component (ViewIssueComponent) sets the updated issue as AssigneeComponent's input issue --> changes are reflected in AssigneeComponent's view

Since these steps are independent, we can test them separately.

One alternative method of testing would be to mock AssigneeComponent's parent component (ViewIssueComponent) in order to test the entire flow, but I don't think this adds much more value (than the current tests)